apps/src/code-studio/assets/show.js

```diff
@@ -13,25 +13,23 @@ var Dialog = require('../dialog');
  *   uploaded.
  * @param onClose {Function} Called when the user closes the asset manager.
  */
-module.exports = function (assetChosen, typeFilter, onClose) {
+module.exports = function (assetChosen, typeFilter, onClose, showUnderageWarning) {
   var codeDiv = document.createElement('div');
   var showChoseImageButton = assetChosen && typeof assetChosen === 'function';
   var dialog = new Dialog({
     body: codeDiv,
     id: 'manageAssetsModal',
     onHidden: onClose
   });
-  const appType = dashboard.project.getStandaloneApp();
-  // TODO: ditch this in favor of react-redux connector
-  // once more of code-studio is integrated into mainline react tree.
-  const studioApp = require('../../StudioApp').singleton;
-  const pageConstants = studioApp.reduxStore.getState().pageConstants;
-  const showWarning = appType === 'applab' && (
-    !pageConstants.isSignedIn || pageConstants.is13Plus
-  );
+
+  if (dashboard.project.getStandaloneApp() === 'applab' && showUnderageWarning === undefined) {
+    const studioApp = require('../../StudioApp').singleton;
+    const pageConstants = studioApp.reduxStore.getState().pageConstants;
+    showUnderageWarning = !pageConstants.isSignedIn || !pageConstants.is13Plus;
```

Comment:
It would be even better if this block lived outside of here. I.e. I think what should happen is that in we make Attachments.jsx a connected component, which allows it to extract isSignedIn/is13Plus from the store (assuming Attachments will always have access to the reduxStore). It can then own all the logic for showUnderageWarning or not.